dqc_us_rules/dqc_us_0018.py

```diff
@@ -0,0 +1,329 @@
+               "docLB": "http://xbrl.fasb.org/us-gaap/2014/us-gaap-2014-01-31.zip/us-gaap-2014-01-31/elts/us-gaap-doc-2014-01-31.xml",  # noqa
+               "entryXsd": "http://xbrl.fasb.org/us-gaap/2014/us-gaap-2014-01-31.zip/us-gaap-2014-01-31/entire/us-gaap-entryPoint-std-2014-01-31.xsd",  # noqa
+            },
+           {
+               "year": 2015,
+               "namespace": "http://fasb.org/us-gaap/2015-01-31",
+               "docLB": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/elts/us-gaap-doc-2015-01-31.xml",  # noqa
+               "entryXsd": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/entire/us-gaap-entryPoint-std-2015-01-31.xsd",  # noqa
+            },
+           {
+               "year": 2016,
+               "namespace": "http://fasb.org/us-gaap/2016-01-31",
+               "docLB": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/elts/us-gaap-doc-2016-01-31.xml",  # noqa
+               "entryXsd": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/entire/us-gaap-entryPoint-std-2016-01-31.xsd",  # noqa
+            },
+           )
+
+
+def _load_cache(val):
+    """
+    Loads the needed deprecated concepts cache into memory
+
+    :param val: ValidateXbrl to load the concepts into
+    :type val: :class:'~arelle.ValidateXbrl.ValidateXbrl'
+    :return: No explicit return, but it loads the deprecated concepts for the
+        taxonomy used for the filing into the deprecated concepts dictionary
+        of the given ValidateXbrl
+    :rtype: None
+    """
+    val.linroleDefinitionIsDisclosure = re.compile(
+        r"-\s+Disclosure\s+-\s", re.IGNORECASE
+    )
+    val.linkroleDefinitionStatementSheet = re.compile(
+        r"[^-]+-\s+Statement\s+-\s+.*", re.IGNORECASE
+    )
+    cntlr = val.modelXbrl.modelManager.cntlr
+    year = _EARLIEST_US_GAAP_YEAR
+
+    for ugt in ugtDocs:
+        ugt_namespace = ugt["namespace"]
+        if ((ugt_namespace in val.modelXbrl.namespaceDocs and
+             len(val.modelXbrl.namespaceDocs[ugt_namespace]) > 0)):
+            val.ugtNamespace = ugt_namespace
+            deprecations_json_file = os.path.join(
+                os.path.dirname(__file__),
+                'resources',
+                'DQC_US_0018',
+                '{}_deprecated-concepts.json'.format(str(year))
+            )
+            file = None
+            try:
+                file = openFileStream(cntlr,
+                                      deprecations_json_file,
+                                      'rt',
+                                      encoding='utf-8')
+
+                val.usgaapDeprecations = json.load(file)
+                file.close()
+
+            except FileNotFoundError:
+                if file:
+                    file.close()
+            return
+        year += 1
+    val.deprecatedFactConcepts = defaultdict(list)
```

Comment:
Not sure if this is intended, but it looks like the only way these three lines get called is if the ugt for loop above completes without ever finding an acceptable ugt_amespace  in the namespaceDocs. Also not quite sure what the point of these are.